File: TwoPointers/TwoSum.java
class TwoSum {

}

class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int front = 0;
        int rear = numbers.length-1;

        while (front < rear) {
            int a = numbers[front];
            int b = numbers[rear];

            if (a == b) {break;}

            if (a + b < target) {
                front++;
                continue;
            }

            rear--;
        }

        return new int[]{++front,++rear};
    }
}

File: ArraysHashing/two_sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        diffs = {}
        for i in range(len(nums)):
            other_val = target - nums[i]
            if other_val in diffs: return [diffs[other_val], i]
            diffs[nums[i]] = i